### ISSUE TITLE
Using Album mode makes the first track too quiet

### ISSUE THREAD

**Issue by Phoenixed** [Sappharad/mp3gainOSX#18]
Noticed this while processing new albums – overall volume is correct but the first track in the list is about 6dB too quiet. Looked at the same files with v1.2.0 and while it says that volume of the first track is correct, it is noticeably too quiet.

**Comment by Sappharad** [comment]
Oops. Someone tried to report this with the beta but didn't provided detailed enough information to reproduce and I thought it was a fluke.

When I introduced album gain, I had to add the list of files on to the commands that I send to mp3gain. But I was already adding on the 'first' file for track gain, so as a result the list of files actually contained the first track twice. So what was actually happening is that the first file had gain applied twice. So if it was supposed to get -3dB gain, it would apply -3dB then apply another -3dB right after that thinking it was modifying the next file.

This is fixed in ff043edbccd2bcccd8982d6111c99f96444a6222 and will be corrected in version 2.0.1 releasing shortly.

As it turns out, the way I handle Album gain is also not the best either. mp3gain doesn't provide track volume information or processing status when running in album mode. And because it has to process all album tracks as a single track, I can't process multiple files at the same time to make it faster. I attempted to solve all of these problems by processing twice for album mode - once to get the track information and a second time to actually apply it. I figured that because it had already processed in track mode and knew the volume for each track that it could then immediately use that to reprocess as album without making you wait very long. Unfortunately, Album mode doesn't actually use the volume data from track mode so it has to re-process all of the files completely again. I will probably not change this behavior though, because then I won't have the original track-specifc volume information to display in the window and I don't want to modify mp3gain + aacgain to give me what I need.
